src/modal/modal.scss

```diff
@@ -67,11 +78,52 @@
         margin-right: $iui-s;
       }
     }
+
+    .iui-modal-content {
+      flex-grow: 2;
+      margin: 0 -#{$iui-m};
+      padding: 0 $iui-m;
+      overflow-y: auto;
+      @supports (overflow-y: overlay) {
+        overflow-y: overlay;
+      }
+    }
   }
+}
 
-  &.iui-modal-visible {
-    visibility: visible;
-    opacity: 1;
-    transition-delay: 0s;
+@mixin iui-modal-full-page {
+  > .iui-modal-dialog {
```

Comment:
We can reduce specificity/nesting by applying the full page modifier on the dialog itself, instead of the parent

Reply:
Good point, but I changed this and it doesn't apply anymore.